fudge-ratio/src/main/java/com/google/sps/servlets/ActivityServlet.java

```diff
@@ -0,0 +1,88 @@
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.PreparedQuery.TooManyResultsException;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.io.IOException;
+import com.google.gson.Gson;
+import com.google.sps.data.Activity;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/data")
+public class ActivityServlet extends HttpServlet {
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+        //Get the name of the activity and the new time taken to complete it
+        String name = request.getParameter("name");
+        String newTimeString = (String)request.getParameter("newTime");
+        Double newTime = Double.parseDouble(newTimeString);
+
+        //Find this activity in datastore
+        Query query = new Query("Activity").setFilter(new FilterPredicate("name", FilterOperator.EQUAL, name));
+        PreparedQuery pq = datastore.prepare(query);
+        Entity result = pq.asSingleEntity();
+
+        //Update number of times that activity was done
+        int numUsed = (int) result.getProperty("numUsed");
+        numUsed = numUsed + 1;
+
+        //Update expected time for activity
+        String expectedTimeString = (String) result.getProperty("expectedTime");
+        Double expectedTime = Double.parseDouble(expectedTimeString);
+        expectedTime = (expectedTime + newTime) / numUsed;
+
+        //Setting properties to entity
+        result.setProperty("expectedTime", expectedTime);
+        result.setProperty("numUsed", numUsed);
+
+        //Posting new entity 
+        datastore.put(result);
+        response.sendRedirect("/index.html");
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Activity");
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Activity> activities = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
```

Comment:
Correct me if I'm wrong, but it seems like we only have one entry per type of task, and you will update that tasks' entry as more people do that task. I thought that we would store an entry for each time any task was done, but if we're going to have one entry per type of task, then I think it would be more useful to also store and update the avg fudge ratio for that task rather than just the expected time. Because ideally we'd want to be able to look up the fudge ratio and use that to make suggestions to the user.  Say the user has a short math assignment that they think will only take them 10 minutes and say the value in the database for avg  math fudge ratio is 1.5. I think it would be more useful to suggest to the user 15 minutes (10 * 1.5) rather than just telling the user what the average expected or actual time is. Let me know if you understood that; I tried my best to explain!

Reply:
Oh got it! Actually I think the first idea was easier to implement since we would not have to update entries (so let me know if you want to change it). But for the time being I guess I will just add the average fudge ratio then

Reply:
now that I think about it,  only storing one entry per task is good because its much less complicated. Since we're returning info based on the average I think this works well

Reply:
The code written makes sense for what we want to do for our data storage. This way works and is easier to send back the suggested time duration we want to show.